### PR TITLE
New: Devdocs!

### PR DESCRIPTION
**Pull request by Bonkles**
This PR sets up the basics for generating developer documentation.  it uses `jsdoc`, the same doc system used by Pixi.js. 

I have made a few small edits to comments in a smattering of files just to eliminate parsing and docgen errors, but the overall content of the documentation is unchanged. 

To use it, run `npm run docs` and you will see a new `dist/doc` folder. 

use `npm start` and wait for the server to come up, then open the index page URL of the documentation: `http://127.0.0.1:8080/dist/docs/index.html`

The page should look like this and contain a searchable index of all the currently documented classes: 

image

### CODE REVIEW

File: modules/core/LocationManager.js [hunk 1]
```diff
@@ -263,7 +263,7 @@ export class LocationManager extends EventEmitter {
    * Returns a locationSetID for a given locationSet (fallback to `+[Q2]`, world)
    * (The locationSet doesn't necessarily need to be resolved to compute its `id`)
    *
-   * @param  `locationSet`  A locationSet Object, e.g. `{ include: ['us'] }`
+   * @param  `locationSet`  A locationSet {Set} Object, e.g. include: [ 'us' ]
```
Comment (bhousel) [inline modules/core/LocationManager.js:266]: This change is not correct

**Comment by bhousel** [comment]
My first thought is it shouldn't get generated under `dist/` - these are built files that get deployed places.
Generated docs should probably go to `docs/`, and we can configure GitHub to make a page for them.

File: .hintrc [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (bhousel) [inline .hintrc:1]: I'd like to know more where this file came from?  Is webhint part of the doc generator?

File: modules/core/LocationManager.js (hunk 1 above)
Comment (Bonkles) [reply modules/core/LocationManager.js:266]: This was one of the 'formatting' issues that jsdoc kept emitting at me- it thinks that the thing in curly braces is a type definition {}. I'll see if I can figure out how to just annotate this as a literal string that it should ignore.

Comment (Bonkles) [reply modules/core/LocationManager.js]: Did this one better- I created a typeDef for the locationSet object in comments, and now jsdoc seems happy.

File: modules/core/LocationManager.js [hunk 3]
```diff
@@ -263,7 +268,7 @@ export class LocationManager extends EventEmitter {
    * Returns a locationSetID for a given locationSet (fallback to `+[Q2]`, world)
    * (The locationSet doesn't necessarily need to be resolved to compute its `id`)
    *
-   * @param  `locationSet`  A locationSet Object, e.g. `{ include: ['us'] }`
+   * @param  `locationSet`  A {LocationSet} Object, e.g. `{ include: ['us'] }`
```
Comment (Bonkles) [inline modules/core/LocationManager.js:271]: Here's the original doc text just with the new LocationSet type mentioned. Did I get the shape of the object correct? Here's how it looks in the docs: 


image

image

File: webdoc.conf.json [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    },
+    "plugin/markdown"  : {
+        "parser"        : "gfm",
+        "hardwrap"      : false
+    },
+    "opts": {
+        "encoding"      : "utf8",
+        "recurse"       : true,
+        "lenient"       : true,
+        "destination"   : "./docs",
```
Comment (Bonkles) [inline webdoc.conf.json:50]: Docs are now going into the 'docs' folder.

**Comment by bhousel** [comment]
This looks good for a start!  We have a _lot_ of work to do to fill out the source docs, and much of the code itself is due for a rewrite, but I am hoping that it gives people some more insight into the shape of things..  😅

**Pull request merged**